utils/stringService.ts

```diff
@@ -75,3 +66,9 @@ export function isStarkDomain(domain: string) {
     domain
   );
 }
+
+export function numberToString(element: number | undefined): string {
+  if (!element) return "";
```

Comment:
Shouldn't you check that element is undefined? Here you might return "" for the integer 0.